activerecord/lib/active_record/relation/batches.rb

```diff
@@ -343,7 +343,9 @@ def batch_condition(relation, columns, values, operators)
         where_clause = predicate_builder[first_clause_column, first_clause_value, operator]
 
         cursor_positions.reverse_each do |column_name, value, operator|
-          where_clause = predicate_builder[column_name, value, operator].and(where_clause)
+          where_clause = predicate_builder[column_name, value, operator == :lteq ? :lt : :gt].or(
```

Comment:
I'll double check it a little bit later but it looks like we are unnecessarily flip-flopping between `gteq/lteq` and `gt/lt` 

For example `in_batches` first builds an array that consists of `gteq/lteq` values
https://github.com/rails/rails/blob/14cfad313765e0b6e28633c363341b145f500063/activerecord/lib/active_record/relation/batches.rb#L310

then immediately passes it to `batch_condition` where we flip it back.
Any chance that we could change `in_batches` on line 310 to build correct values from the beginning? Similar thing seems to be happening in  `apply_start_limit` and `apply_finish_limit` - we build an array with `gteq/lteq` just to flip it back later in `batch_condition`

Though I may be missing some corner cases and I have to admit it's a little hard to follow the clause building logic just by reading it. I'll try to have a closer look later. But at the end of the day it's a tiny performance concern rather than a bug